File: database/database-annotation-processor/src/test/java/io/koraframework/database/common/annotation/processor/repository/error/UnknownEntityField.java
package io.koraframework.database.common.annotation.processor.repository.error;

import io.koraframework.database.common.annotation.Query;
import io.koraframework.database.common.annotation.Repository;
import io.koraframework.database.jdbc.JdbcRepository;

@Repository
public interface UnknownEntityField extends JdbcRepository {

    @Query("SELECT * FROM table WHERE name = :dto.name")
    String wrongEntityField(Dto dto);

    record Dto(long id) {}
}

File: database/database-symbol-processor/src/test/kotlin/io/koraframework/database/symbol/processor/repository/error/UnknownEntityField.kt
package io.koraframework.database.symbol.processor.repository.error

import io.koraframework.database.common.annotation.Query
import io.koraframework.database.common.annotation.Repository
import io.koraframework.database.jdbc.JdbcRepository

@Repository
interface UnknownEntityField : JdbcRepository {

    @Query("SELECT * FROM table WHERE name = :dto.name")
    fun wrongEntityField(dto: Dto): String

    data class Dto(val id: Long)
}
